tests/app/main/test_frameworks.py

```diff
@@ -4284,6 +4277,30 @@ def test_upload_signature_page(
         assert res.status_code == 302
         assert res.location == 'http://localhost/suppliers/frameworks/g-cloud-8/234/contract-review'
 
+    @mock.patch('dmutils.s3.S3')
+    @mock.patch('app.main.views.frameworks.file_is_empty')
```

Comment:
I don't think we need to be mocking `file_is_empty` given that we shouldn't reach that part of logic in the view?